### ISSUE TITLE
Failure with completion for read() on ZipFile: "RuntimeError: maximum recursion depth exceeded"

### ISSUE THREAD

**Issue by blueyed** [davidhalter/jedi#574]
I've tried to complete after `read()` of a `zipfile.Zipfile`, which took very long:

```
import jedi
s="import zipfile; z = zipfile.ZipFile(\"foo\"); z.read(n)."
jedi.Script(s, 1, len(s), ".").completions()
```

Using `strace`, it looks like it tries to open everything / a lot from the import path?!

```
% strace -f -eopen python -c 'import jedi; s="import zipfile; z = zipfile.ZipFile(\"foo\"); z.read(n)."; print(jedi.Script(s, 1, len(s), ".").completions())'
```

In the end it aborts with `RuntimeError: maximum recursion depth exceeded`.

I hope I got the arguments to `jedi.Script` right (`len(s)`).

**Comment by davidhalter** [comment]
You always seem to find the strange cases that are hard to fix :D Thanks though for for debugging it a bit. I've fixed a few things along the lines and I've limited the dynamic parameter completion to not go completely recursively crazy.

However it takes still 5 seconds to complete (even with disabled dynamic parameters it takes 2.5s). This is not good but way better than RecursionErrors and times of 30s plus.

I'm leaving this open for a bit. However I doubt that I'm going to fix it very soon.
